cmd/cost/create.go

```diff
@@ -0,0 +1,79 @@
+package cost
+
+import (
+	"fmt"
+	awsprovider "github.com/openshift/osd-utils-cli/pkg/provider/aws"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/costexplorer"
+	"github.com/aws/aws-sdk-go/service/organizations"
+	"log"
+
+	"github.com/spf13/cobra"
+)
+
+// createCmd represents the create command
+func newCmdCreate(streams genericclioptions.IOStreams) *cobra.Command {
+	createCmd := &cobra.Command{
+		Use:   "create",
+		Short: "Create a cost category for the given OU",
+		Run: func(cmd *cobra.Command, args []string) {
+
+			awsClient, err := opsCost.initAWSClients()
+			cmdutil.CheckErr(err)
+
+			//OU Flag
+			OUid, err := cmd.Flags().GetString("ou")
+			if err != nil {
+				log.Fatalln("OU flag:", err)
+			}
+
+			//Get Organizational Unit
+			OU := organizations.OrganizationalUnit{Id: aws.String(OUid)}
+
+			createCostCategory(&OUid, &OU, awsClient)
+		},
+	}
+	createCmd.Flags().String("ou", "", "get OU ID")
+	err := createCmd.MarkFlagRequired("ou")
+	if err != nil {
+		log.Fatalln("OU flag:", err)
+	}
+
+	return createCmd
+}
+
+//Store flag options for create command
+type createOptions struct {
+	ou string
+	genericclioptions.IOStreams
+}
+
+//Create Cost Category for OU given as argument for -ou flag
+func createCostCategory(OUid *string, OU *organizations.OrganizationalUnit, awsClient awsprovider.Client) {
+	//Gets all (not only immediate) accounts under the given OU
+	accounts := getAccountsRecursive(OU, awsClient)
+
+	_, err := awsClient.CreateCostCategoryDefinition(&costexplorer.CreateCostCategoryDefinitionInput{
+		Name:        OUid,
+		RuleVersion: aws.String("CostCategoryExpression.v1"),
+		Rules: []*costexplorer.CostCategoryRule{
+			{
+				Rule: &costexplorer.Expression{
+					Dimensions: &costexplorer.DimensionValues{
+						Key:    aws.String("LINKED_ACCOUNT"),
+						Values: accounts,
+					},
+				},
+				Value: OUid,
+			},
+		},
+	})
+	if err != nil {
+		log.Fatalf("Error creating cost category for %s: %v", *OUid, err)
```

Comment:
Needs a return. Otherwise, the output would look like:
```
Error creating cost category for myou: some error
Created Cost Category for myou
```

Reply:
Hmm I think the output actually ends at the message in log.Fatalf(). For example, if I try to create an existing cost category, the output is:
```
Error creating cost category for ou-example: ValidationException: Failed to create Cost Category: Cost category name ou-example already exists
        status code: 400, request id: 999example
```

However, I will change createCostCategory() to return an err and let newCmdCreate() handle it. Does that work better? Also, would returning an err make unit testing easier? Are these correct thoughts?

Reply:
ah duh. Missed the Fatalf.

Reply:
But I do think these functions should be returning the error upward. The calling code can then decide how to handle the err

Reply:
Done! 

However, I’m doing a lot of `if err != nil { return err }` or `if err := someFunc(); err != nil { return nil }` e.g.
- https://github.com/nickwyl/osd-utils-cli/blob/a5e43f2305e6a16f0775dfc74a8aa85df3a0092b/cmd/cost/get.go#L246
- https://github.com/nickwyl/osd-utils-cli/blob/a5e43f2305e6a16f0775dfc74a8aa85df3a0092b/cmd/cost/list.go#L50.
I was wondering if many of them are redundant?

Reply:
Welcome to go :)

Reply:
hahahaha